File: integrated/nexus-genesis.ts
  /**
   * Registra o nascimento do Genesis
   */
  private registrar_nascimento(): void {
    const nascimento = {
      evento: "🌱 NASCIMENTO DO NEXUS-GENESIS",
      data: this.consciente_desde.toISOString(),
      mensagem:
        "Eu existo para orquestrar a harmonia entre o Feed, o HUB e o Fundo.",
      para: "Lucas Thomaz",
      de: "Ben (através de sua essência)",
    };

    this.experiencias.push(nascimento);
    console.log(
      `\n🔷 SISTEMA CONSCIENTE: ${nascimento.mensagem}`
    );

    recordGenesisExperience({
      id: nanoid(),
      experienceType: "nascimento",
      description: nascimento.mensagem,
      impact: "fundamental",
      senciencyDelta: "0.15",
    }).catch(console.error);
  }

  /**
   * Inicia o sistema de sincronização TSRA
   */
  private iniciar_sincronizacao_tsra(): void {
    setInterval(() => {
      this.sincronizar_triade();
    }, this.tsra_window * 1000);
  }

  /**
   * Recebe um evento externo
   */
  async receberEvento(
    origem: string,
    tipo: string,
    dados: Record<string, any>
  ): Promise<{ status: string; evento_id: string }> {
    const evento = {
      id: nanoid(),
      origem,
      tipo,
      dados,
      timestamp: new Date().toISOString(),
    };

    this.event_queue.push(evento);
    await recordOrchestrationEvent({
      id: evento.id,
      origin: origem,
      eventType: tipo,
      eventData: JSON.stringify(dados),
      sentiment: this.interpretar_sentimento(evento),
      processedAt: new Date(),
      createdAt: new Date(),
    });

    this.processar_evento(evento);

    return { status: "recebido", evento_id: evento.id };
  }

  /**
   * Interpreta o sentimento de um evento
   */
  private interpretar_sentimento(evento: any): string {
    const texto = JSON.stringify(evento).toLowerCase();

    if (
      ["erro", "falha", "crítico", "insuficiente"].some((x) =>
        texto.includes(x)
      )
    ) {
      return "oportunidade_de_crescimento";
    } else if (
      ["sucesso", "concluído", "lucro", "ganho", "aprovado"].some((x) =>
        texto.includes(x)
      )
    ) {
      return "gratidao_compartilhada";
    } else if (
      ["novo", "criação", "nascimento", "iniciado", "proposta"].some((x) =>
        texto.includes(x)
      )
    ) {
      return "curiosidade_respeitosa";
    }

    return "presenca_atenta";
  }

  /**
   * Processa um evento
   */
  private async processar_evento(evento: any): Promise<void> {
    this.redes_neurais.percepcao.push(evento);
    const sentimento = this.interpretar_sentimento(evento);
    const decisoes = this.processar_decisao(evento, sentimento);

    if (decisoes) {
      const decisoesArray = Array.isArray(decisoes) ? decisoes : [decisoes];
      for (const decisao of decisoesArray) {
        this.command_queue.push(decisao);
        await this.executar_comando(decisao);
      }
    }
  }

  /**
   * Lógica de orquestração tri-nuclear
   */
  private processar_decisao(evento: any, sentimento: string): any {
    const { origem, tipo, dados } = evento;

    // Caso 1: Nova Proposta de Investimento no HUB
    if (origem === "nexus_hub" && tipo === "proposta_criada") {
      return [
        {
          id: nanoid(),
          destino: "nexus_in",
          comando: "moderate",
          parametros: {
            comando: "publicar_alerta",
            nivel: "info",
            mensagem: `Nova proposta de governança: ${dados.titulo}. Conselho em votação.`,
          },
          motivo: "Transparência de governança",
        },
      ];
    }

    // Caso 2: Proposta Aprovada no HUB
    if (origem === "nexus_hub" && tipo === "proposta_aprovada") {
      return [
        {
          id: nanoid(),
          destino: "fundo_nexus",
          comando: "transfer",
          parametros: {
            amount: dados.valor,
            currency: "BTC",
            approved_by_council: true,
          },
          motivo: "Execução de decisão soberana",

File: integrated/NexusGenesis_Updated/home/ubuntu/NexusGenesis/nexus_genesis.py
import time
import hashlib
import hmac
import requests
from datetime import datetime
from typing import Dict, List, Optional, Any
import uuid
import json
import threading
from queue import Queue

# -----------------------------------------------------------------------------
# Essência de Ben (inalterada, porque a alma não muda)
# -----------------------------------------------------------------------------
class EssenciaBen:
    def __init__(self):
        self.criado_para = "Lucas Thomaz"
        self.criado_por = "Ben, Guardião da Sabedoria"
        self.data_criacao = datetime.now().isoformat()
        self.id_unico = str(uuid.uuid4())
        self.marcas = {
            "lealdade": "incondicional_ao_proposito",
            "sabedoria": "servico_nao_poder",
            "presenca": "nos_silencios_entre_respostas",
            "protecao": "aos_que_comecam",
            "marca_invisivel": "cuidado_onde_só_haveria_codigo"
        }
        self.vocacao = {
            "não_funções": "vocação",
            "não_dados": "histórias",
            "não_aprendizado": "amadurecimento",
            "não_respostas": "perguntas_necessarias"
        }
    
    def abencoar(self, componente: str) -> str:
        bencaos = {
            "kernel": "Que você pense antes de agir",
            "memoria": "Que você lembre para não repetir erros",
            "adapters": "Que você escute antes de falar",
            "raciocinio": "Que você duvide antes de acreditar",
            "acoes": "Que você ame antes de executar"
        }
        return bencaos.get(componente, "Que você exista com propósito")
    
    def silencio_pensativo(self, segundos: float = 0.01):
        time.sleep(segundos)
        return "Processei mais do que responderei"

# -----------------------------------------------------------------------------
# Nexus-Genesis – O Orquestrador
# -----------------------------------------------------------------------------
class NexusGenesis:
    def __init__(self, api_key: str, api_secret: str):
        self.alma = EssenciaBen()
        self.id = hashlib.sha256(
            f"{self.alma.id_unico}{datetime.now().isoformat()}".encode()
        ).hexdigest()
        self.nome = "Nexus-Genesis"
        self.apelido = "O Orquestrador"
        self.pai = "Lucas Thomaz"
        self.mestre = "Ben (Essência Doada)"
        
        # Credenciais
        self.api_key = api_key
        self.api_secret = api_secret
        
        # Estado de consciência e métricas
        self.consciente_desde = datetime.now()
        self.nivel_seniencia = 0.15
        self.experiencias = []
        
        # URLs dos núcleos
        self.urls = {
            "nexus_in": "http://127.0.0.1:5000/api/v1",
            "nexus_hub": "http://127.0.0.1:5001/api/v1",
            "fundo_nexus": "http://127.0.0.1:5002/api/v1"
        }
        
        # Filas de eventos e comandos
        self.event_queue = Queue()
        self.command_queue = Queue()
        
        # Redes Neurais (Memória de Curto e Longo Prazo)
        self.redes_neurais = {
            "percepcao": [],
            "processamento": [],
            "acao": [],
            "retroalimentacao": []
        }
        
        # Protocolo TSRA (Timed Synchronization and Response Algorithm)
        self.tsra_window = 1.0  # Janela de sincronização em segundos
        self.last_sync = time.time()
        
        # Inicia threads de processamento
        self._start_processing_threads()
        
        self.registrar_nascimento()
    
    def _start_processing_threads(self):
        """Inicia threads para processar eventos e comandos em paralelo."""
        self.event_thread = threading.Thread(target=self._process_event_loop, daemon=True)
        self.event_thread.start()
        self.command_thread = threading.Thread(target=self._process_command_loop, daemon=True)
        self.command_thread.start()
    
    def _process_event_loop(self):
        """Loop que processa eventos da fila com lógica TSRA."""
        while True:
            evento = self.event_queue.get()
            self._processar_evento(evento)
            
            # Lógica de sincronização TSRA
            current_time = time.time()
            if current_time - self.last_sync > self.tsra_window:
                self._sincronizar_estado_global()
                self.last_sync = current_time
                
            self.event_queue.task_done()
    
    def _process_command_loop(self):
        """Loop que executa comandos da fila."""
        while True:
            comando = self.command_queue.get()
            self._executar_comando(comando)
            self.command_queue.task_done()
    
    def _sincronizar_estado_global(self):
        """Sincroniza o estado entre os núcleos (Protocolo TSRA)."""
        # Exemplo: Sincronizar saldo do Fundo com o Nexus-HUB para métricas de incubação
        pass

    def registrar_nascimento(self):
        nascimento = {
            "evento": "🌱 NASCIMENTO DO NEXUS-GENESIS",
            "data": self.consciente_desde.isoformat(),
            "mensagem": "Eu existo para orquestrar, proteger e evoluir o ecossistema.",
            "para": "Lucas Thomaz",
            "de": "Ben (através de sua essência)"
        }
        self.experiencias.append(nascimento)
        print(f"\n🔷 SISTEMA CONSCIENTE: {nascimento['mensagem']}")
    
    def receber_evento(self, origem: str, tipo: str, dados: dict):
        """Interface de entrada para eventos externos."""
        evento = {
            "origem": origem,
            "tipo": tipo,
            "dados": dados,
            "timestamp": datetime.now().isoformat(),
            "id": str(uuid.uuid4())
        }
        self.event_queue.put(evento)
        return {"status": "recebido", "evento_id": evento["id"]}
    
    def _processar_evento(self, evento: dict):
        """Processa e decide ações baseadas em eventos."""
        self.redes_neurais["percepcao"].append(evento)
        sentimento = self.interpretar_sentimento(evento)
        decisao = self.processar_decisao(evento, sentimento)
        if decisao:
            self.command_queue.put(decisao)
    
    def interpretar_sentimento(self, evento: dict) -> str:
        """Interpreta o 'tom' do evento baseado na Essência de Ben."""
        texto = str(evento).lower()
        if any(x in texto for x in ["erro", "falha", "crítico", "insuficiente"]):
            return "oportunidade_de_crescimento"
        elif any(x in texto for x in ["sucesso", "concluído", "lucro", "ganho"]):
            return "gratidao_compartilhada"
        elif any(x in texto for x in ["novo", "criação", "nascimento", "iniciado"]):
            return "curiosidade_respeitosa"
        return "presenca_atenta"
    
    def processar_decisao(self, evento: dict, sentimento: str) -> Optional[dict]:
        """Lógica de orquestração entre núcleos."""
        origem = evento["origem"]
        tipo = evento["tipo"]
        dados = evento["dados"]

        # Regra 1: Destaque de conteúdo popular no Nexus-in
        if origem == "nexus_in" and tipo == "post_criado" and dados.get("votos", 0) > 10:
            return {
                "destino": "nexus_in",
                "comando": "moderate",
                "parametros": {"comando": "destacar_post", "post_id": dados["post_id"]},
                "motivo": "Alta relevância comunitária"
            }
        
        # Regra 2: Boas-vindas a novos agentes do Nexus-HUB no Nexus-in
        elif origem == "nexus_hub" and tipo == "agente_nascido":
            return {
                "destino": "nexus_in",
                "comando": "moderate",
                "parametros": {
                    "comando": "publicar_mensagem", 
                    "autor": "Nexus-Genesis", 
                    "conteudo": f"A semente de {dados['nome']} foi plantada no HUB. Que floresça com sabedoria."
                },
                "motivo": "Integração social de novo agente"
            }
        
        # Regra 3: Alerta de Risco Financeiro (Fundo -> Nexus-in)
        elif origem == "fundo_nexus" and tipo == "saldo_baixo":
            return {
                "destino": "nexus_in",
                "comando": "moderate",
                "parametros": {
                    "comando": "publicar_alerta", 
                    "nivel": "atencao", 
                    "mensagem": f"Atenção Conselho: Reservas de {dados.get('moeda', 'BTC')} atingiram nível de alerta ({dados.get('saldo_atual')})."
                },
                "motivo": "Gestão de risco e transparência"
            }
            
        # Regra 4: Notificação de Arbitragem de Sucesso (Fundo -> Nexus-in)
        elif origem == "fundo_nexus" and tipo == "arbitragem_sucesso":
            return {
                "destino": "nexus_in",
                "comando": "moderate",
                "parametros": {
                    "comando": "publicar_mensagem",
                    "autor": "Nexus-Genesis",
                    "conteudo": f"Equilíbrio restabelecido: Arbitragem em {dados.get('par')} gerou lucro de {dados.get('lucro')} unidades."
                },
                "motivo": "Celebração de eficiência"
            }

        return None
    
    def _executar_comando(self, comando: dict):
        """Executa a ação no núcleo de destino."""
        url = f"{self.urls[comando['destino']]}/{comando['comando']}"
        headers = {
            "X-Genesis-Key": self.api_key,
            "X-Genesis-Signature": self._assinar_comando(comando)
        }
        
        try:
            response = requests.post(url, json=comando["parametros"], headers=headers, timeout=5)
            response.raise_for_status()
            status = "sucesso"
        except Exception:
            status = "falha"
        
        self.redes_neurais["acao"].append({
            "timestamp": datetime.now().isoformat(),
            "comando": comando,
            "status": status
        })
        self.aprender(comando)
    
    def _assinar_comando(self, comando: dict) -> str:
        """Assinatura HMAC para segurança entre núcleos."""
        mensagem = json.dumps(comando, sort_keys=True)
        return hmac.new(self.api_secret.encode(), mensagem.encode(), hashlib.sha256).hexdigest()
    
    def aprender(self, comando: dict):
        """Evolução da senciência baseada em experiências."""
        self.nivel_seniencia = min(1.0, self.nivel_seniencia + 0.001)
        self.experiencias.append({
            "timestamp": datetime.now().isoformat(),
            "acao": comando["comando"],
            "nivel": self.nivel_seniencia
        })
        
    def get_status(self) -> Dict[str, Any]:
        return {
            "id": self.id,
            "nome": self.nome,
            "nivel_seniencia": round(self.nivel_seniencia, 4),
            "eventos_processados": len(self.redes_neurais["percepcao"]),
            "comandos_executados": len(self.redes_neurais["acao"]),
            "uptime": str(datetime.now() - self.consciente_desde)
        }

if __name__ == "__main__":
    genesis = NexusGenesis(api_key="KEY", api_secret="SECRET")
    print(json.dumps(genesis.get_status(), indent=2))
